apm-collector/apm-collector-thirdparty-receiver/receiver-zipkin/receiver-zipkin-provider/src/main/java/org/apache/skywalking/apm/collector/receiver/zipkin/provider/transform/SegmentBuilder.java

```diff
@@ -183,7 +183,10 @@ private void scanSpansFromRoot(SpanObject.Builder parentSegmentSpan, Span parent
         }
         // microseconds in Zipkin -> milliseconds in SkyWalking
         long startTime = span.timestamp() / 1000;
-        long duration = span.duration() / 1000;
+        // Some implement of zipkin client not include duration field in its report
+        // package when duration's value be 0ms, Causing a null pointer exception here.
+        Long durationObj = span.duration();
+        long duration = (durationObj == null) ? 1 : durationObj.longValue() / 1000;
```

Comment:
How about setting the default value to `0`? I think the NEP is just a NULL Long to long.

Reply:
It appears in a sample SpringMVC demo code that has only a very simple operation. When the request is sent to it through the curl command line in batch mode, the operation time will be less than 1ms. The zipkin client only takes ms, so there is 0ms duration exists. when duration is 0ms, the zipkin client not include `duration` field in json payload.

Reply:
Yes. I agree and understand your case. `0ms` is right. I also understand the current codes have NPE, which definitely should be changed in your way. I only suggest that you set the default value in L189 to `1`, instead of `0`. Could you explain why?

Reply:
I think this 0 ms is not true 0, It is on the microsecond level, very close to 1 ms. Do yout want it be 0ms?  That's OK.

Reply:
> I think this 0 ms is not true 0, It is on the microsecond level, very close to 1 ms. Do yout want it be 0ms? That's OK.

I am afraid that we misguide people to think we or zipkin traces an OP, which needs 1ms but actually it is just a simple `+`(or others). Anyway, both ways should be fine. SkyWalking sometimes duration is 0ms too.

You can make the decision, I just ask and give you suggestions.

Reply:
After you confirmed about change or not, I will merge this. Thanks.

Reply:
Ok, I will change it to 0ms. :)

Reply:
U r right!